test/unit/component/parser/json/BpmnJsonParser.sub.process.test.ts

```diff
@@ -78,12 +83,140 @@ describe('parse bpmn as json for sub-process', () => {
+          `should convert as Shape, when a process contains a ${expandedKind} ${bpmnSubProcessKind} sub-process with $input attribute as $title`,
+          ({ title, input, expectedAttribute }: { title: string; input: keyof TFlowNode; expectedAttribute: keyof ExpectedShape }) => {
+            const inputString = String(input);
+            const json = buildDefinitions({
+              process: {
+                subProcess: {
+                  id: 'sub_process_id_0',
+                  triggeredByEvent: triggeredByEvent,
+                  [input]: title === 'array' ? [`flow_${inputString}_1`, `flow_${inputString}_2`] : `flow_${inputString}_1`,
+                  isExpanded,
+                },
+              },
+            });
+
+            const model = parseJsonAndExpectOnlyFlowNodes(json, 1);
+
+            verifyShape(model.flowNodes[0], {
+              shapeId: 'shape_sub_process_id_0',
+              bpmnElementId: 'sub_process_id_0',
+              bpmnElementName: undefined,
+              bpmnElementKind: ShapeBpmnElementKind.SUB_PROCESS,
+              bpmnElementMarkers: expectedBpmnElementMarkers,
+              bounds: { x: 67, y: 23, width: 456, height: 123 },
+              [expectedAttribute]: title === 'array' ? [`flow_${inputString}_1`, `flow_${inputString}_2`] : [`flow_${inputString}_1`],
+            });
+          },
+        );
+
+        it.each`
+          title         | flowKind          | expectedAttribute
+          ${'incoming'} | ${'sequenceFlow'} | ${'bpmnElementIncomingIds'}
+          ${'outgoing'} | ${'sequenceFlow'} | ${'bpmnElementOutgoingIds'}
+          ${'incoming'} | ${'association'}  | ${'bpmnElementIncomingIds'}
+          ${'outgoing'} | ${'association'}  | ${'bpmnElementOutgoingIds'}
+        `(
+          `should convert as Shape, when a process contains a ${expandedKind} ${bpmnSubProcessKind} sub-process with $title $flowKind`,
+          ({ title, flowKind, expectedAttribute }: { title: string; flowKind: keyof TFlowNode; expectedAttribute: keyof ExpectedShape }) => {
+            const json = buildDefinitions({
+              process: {
+                subProcess: { id: 'sub_process_id_0', triggeredByEvent: triggeredByEvent, isExpanded },
+                [flowKind]: {
+                  id: `flow_${title}`,
+                  sourceRef: title === 'incoming' ? 'unknown' : 'sub_process_id_0',
+                  targetRef: title === 'incoming' ? 'sub_process_id_0' : 'unknown',
+                },
+              },
+            });
+
+            const model = parseJsonAndExpectOnlyEdgesAndFlowNodes(json, 1, 1);
+
+            verifyShape(model.flowNodes[0], {
+              shapeId: 'shape_sub_process_id_0',
+              bpmnElementId: 'sub_process_id_0',
+              bpmnElementName: undefined,
+              bpmnElementKind: ShapeBpmnElementKind.SUB_PROCESS,
+              bpmnElementMarkers: expectedBpmnElementMarkers,
+              bounds: { x: 67, y: 23, width: 456, height: 123 },
```

Comment:
**suggestion:** duplicated everywhere, can we introduce a `expectedSubProcessBounds` variable?
Same rationale as "flow nodes" tests